previews/cosmosdb/transactions.md

```diff
@@ -0,0 +1,61 @@
+---
+title: Transactions in Azure Cosmos DB persistence
+component: cosmosdb
+related:
+- nservicebus/outbox
+reviewed: 2020-09-11
+---
+
+By default, the persister does not attempt to atomically commit saga data and/or business data. Through the use of the [Cosmos DB transactional batch API](https://devblogs.microsoft.com/cosmosdb/introducing-transactionalbatch-in-the-net-sdk/), saga data and/or business data can be atomically committed if everything is stored in the same partition within a container.
+
+A custom ['Behavior'](/nservicebus/pipeline/manipulate-with-behaviors.md) must be introduced to identify and insert the [`PartitionKey`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.documents.partitionkey?view=azure-dotnet) value into the pipeline context for use by storage operations that occur during the processing of a given message.
+
+WARN: Do not use a [message mutator](/nservicebus/pipeline/message-mutators.md) to identify the PartitionKey. Message mutators do not offer the necessary control or timing to reliably interact with this persistance.
+
+There are 2 options for which stage to choose to introduce the custom behavior:
+
+## `ITransportReceiveContext` stage
+
+This is the earliest stage in the message processing pipeline. At this stage only the headers and a byte array representation of the message body are available. 
```

Comment:
```suggestion
This is the earliest stage in the message processing pipeline. At this stage only the message ID, the headers and a byte array representation of the message body are available. 
```

I've explicitly added a message ID as it's available as a property at the physical stage. Unlike the logical stage, where it's only available via headers.